RELEASING.adoc

```diff
@@ -0,0 +1,92 @@
+= Releasing Kiali UI
+
+This document covers the specifics of releasing Kiali UI, and is meant as a
+guide for Kiali release managers.
+
+The main document covering core can be found link:https://github.com/kiali/kiali/blob/master/RELEASING.adoc[here].
+
+== Requirements
+
+You must have write permissions to the public GitHub repository in order to be
+able to push the tags, but that's pretty much it. Most of our release process
+is done automatically by Travis.
+
+This document also uses 1.0.0 as the new version, just as an example.
+
+== Preparing for a new version
+
+After a release happens, the first step for a new "batch" of modifications is
+to update link:package.json[] with the new version, and creating a PR for it:
+
+[source, diff]
+----
+diff --git a/package.json b/package.json
+index 7f699b1..08bc0a8 100644
+--- a/package.json
++++ b/package.json
+@@ -1,6 +1,6 @@
+ {
+   "name": "@kiali/kiali-ui",
+-  "version": "0.5.1",
++  "version": "1.0.0",
+   "bugs": {
+     "url": "https://issues.jboss.org/projects/KIALI/issues/"
+   },
+----
+
+This is to signalize everyone that everything that goes into master from that
+point forward belongs to that version. Note that this is supposed to be the
+same version that is going to be released for the core, so they are in sync.
+
+The versioning scheme that we are following is similar to
+link:http://semver.org[semver], but not as strict, given both that we are not
+1.0 yet and that we are the end product, in the sense that nothing should
+depend on us to do anything. For now, and that might change in the future, our
+API is private and we make no guarantees that compatibility is going to be
+mantained.
+
+For our versioning scheme, the `major` part means a major release, and that's
+decided in advance by the team. The `minor` means new features are included
+with the release, and the `patch` version covers only bugfixes.
+
+The UI and the core might be out of sync in `patch` versions, but they should
+be in sync for `minor`. That means that, if the ui is 0.5.10, then any core
+version on 0.5.X should work with it, but that are no guarantees that it will
+work with core 0.4.X or core 0.6.X.
+
+Of course, that's a soft limitation, and your mileage may vary. Sometimes it
+might work just fine, we just don't give any guarantees that it will.
+
+== Creating the Tag
+
+After the work that goes into a new version is done, the next step is to create
+the tag and push it. Everything else will be taken care by Travis.
+
+[source, bash]
+----
+$ git tag v1.0.0 <hash-of-last-commit-of-release>
```

Comment:
A more `npm` automated approach could be
```
npm version minor
npm push && npm push --tags
```
Npm will automatically increment the version based on `patch|minor|major'

Reply:
+1
but with yarn :)
https://yarnpkg.com/lang/en/docs/cli/version/

Reply:
I'm up for doing that, but tbh I'd prefer if this was added after someone did a release with it. I don't know how it works, and do not have the permissions to do a release on it.

I think this is outside of this scope, but can be added after the next release.